estatioapp/app/src/main/java/org/estatio/module/application/imports/contributed/Budget_importExportBudget.java

```diff
@@ -33,11 +27,15 @@
 import org.estatio.module.budget.dom.budget.Budget;
 
 // TODO: need to untangle this and push back down to budget module
-@DomainService(
-        nature = NatureOfService.VIEW_CONTRIBUTIONS_ONLY
-)
+@Mixin(method = "act")
 public class Budget_importExportBudget {
 
+    private final Budget budget;
+
+    public Budget_importExportBudget(Budget budget) {
+        this.budget = budget;
+    }
+
     @PostConstruct
```

Comment:
remove this init() method, it isn't something that mixins support.  Actually, perhaps it could instead be refactored into a disableAct(), ie:

    public String disableAct() { return excelService==null ? "Requires ExcelService to be configured" : null; }

Reply:
Maybe leave out altogether? I think I do not do this check anymore when writing from scratch...

Reply:
yes, that's probably true.  I think I was just trying to work out what it might map onto.  But yes, we don't really need this check at all.